### ISSUE TITLE
SDL2 branch does not show cursor

### ISSUE THREAD

**Issue by stephanemagnenat** [Globulation2/glob2#40]
Testing the SDL2 branch here (Ubuntu 20.04 (Kubuntu), X11, Nvidia RTX 2060 super, proprietary driver version 450.102.04), I get the following errors.

- By default (with software renderer), I see no cursor, the window has refresh problem, and this on the console:
  ```
  Toolkit : can't set screen to 1280x720 at 32 bpp
  Toolkit : Can't create renderer for NULL surface
  Toolkit : Can't set screen resolution, resetting to default of 800x600
  ```
- With hardware (`-g` flag), Glob2 segfaults and the console shows the following problems:
  ```
  Toolkit : can't set screen to 1280x720 at 32 bpp
  Toolkit : No hardware accelerated renderers available
  Toolkit : Can't set screen resolution, resetting to default of 800x600
  Toolkit : can't set screen to 800x600 at 32 bpp
  Toolkit : No hardware accelerated renderers available
  ```

The SDL 1.2 version works fine with this setup.

**Comment by stephanemagnenat** [comment]
After investigation, it seems that the changes to the resolution settings are broken. Code to set the resolution was added in the constructor of `GraphicContext`, and then the call to `setRes` is done which attempts to reset the resolution, this with doubtful order.  I think the proper way to do this is as before, only do the window creation and resolution setting in `setRes`, and do it differently whether the window exists already or not.

**Comment by stephanemagnenat** [comment]
This is fixed by f03d62f.
